### PR TITLE
Add simple `PFCoilSupportBuilder` (updated)

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

#Requires #1942

image
Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
Still need to add mroe tests but based off of #1942

File: tests/builders/test_coil_supports.py [hunk 1]
```diff
@@ -86,3 +89,90 @@ def test_good_support_radius(self, tf, x_gs):
...
+class TestPFCoilSupportBuilder:
+
+    my_test_params = PFCoilSupportBuilderParams(
+        tf_wp_depth=1.4,
+        tf_wp_width=0.8,
+        tk_tf_side=0.05,
+        pf_s_tk_plate=0.15,
+        pf_s_n_plate=4,
+        pf_s_g=0.05,
+    )
```
Comment (je-cook) [inline tests/builders/test_coil_supports.py:103]: ParameterFrame needs setting up as in #1942, I will get round to fixing #1947 soon...

File: bluemira/builders/coil_supports.py [hunk 2]
```diff
@@ -276,3 +279,249 @@ def build_xyz(
...
+                dist = np.hypot(dx, dz)
+                distances.append(dist)
+                directed_intersections.append(inter)
+
+        if len(directed_intersections) > 0:
+            i_min = np.argmin(distances)
+            p_inter = directed_intersections[i_min]
+            return p_inter
+        else:
+            return None
```
Comment (je-cook) [inline bluemira/builders/coil_supports.py:404]: This automatically happens as there is nothing past this point
```suggestion
```

File: bluemira/builders/coil_supports.py [hunk 3]
```diff
@@ -276,3 +279,249 @@ def build_xyz(
...
+
+        directed_intersections = []
+        for inter in intersections:
+            direction = inter - point
+            direction /= np.linalg.norm(direction)
+            if np.dot(correct_direction, direction) < 0:
+                pass
+            else:
+
```
Comment (je-cook) [inline bluemira/builders/coil_supports.py:391]:
```suggestion
            if not np.dot(correct_direction, direction) < 0:
```

**Review by je-cook — CHANGES_REQUESTED** [review]
Looks good will run now just to see locally, a few little things

**Review by je-cook — APPROVED** [review]

**Pull request merged**